src/test/java/PlanMailServletTest.java

```diff
@@ -0,0 +1,221 @@
+        .thenReturn(MESSAGES);
+    Mockito.when(planMailResponseHelper.getWordCount(Mockito.any())).thenReturn(TEST_WORD_COUNT);
+    PlanMailResponse actual = getServletResponse();
+    List<Date> startHour = Arrays.asList(new Date(2020 - OFFSET_YEAR, 4, 19, 10, 0));
+    List<Date> endHour = Arrays.asList(new Date(2020 - OFFSET_YEAR, 4, 19, 10, 10));
+    List<DateInterval> meetingTimes = new ArrayList<>();
+    for (int index = 0; index < startHour.size(); index++) {
+      meetingTimes.add(new DateInterval(startHour.get(index), endHour.get(index)));
+    }
+    PlanMailResponse expected = new PlanMailResponse(500, 50, 10, meetingTimes);
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void alreadyCreatedEvent() throws Exception {
+    // Test case where there are unread emails that will take 10 minutes to read
+    // But an event (event 2) has already been created to go through them
+    // We should get no proposed potential event
+    Mockito.when(calendarClient.getCalendarList()).thenReturn(TWO_CALENDARS);
+    Mockito.when(calendarClient.getUpcomingEvents(PRIMARY, CURRENT_TIME, END_TIME))
+        .thenReturn(EVENT_ONE);
+    Mockito.when(calendarClient.getUpcomingEvents(SECONDARY, CURRENT_TIME, END_TIME))
+        .thenReturn(EVENT_TWO);
+    Mockito.when(calendarClient.getCurrentTime()).thenReturn(CURRENT_TIME);
+    Mockito.when(gmailClient.getUnreadEmailsFromNDays(MESSAGE_FORMAT, NUMBER_DAYS_UNREAD))
+        .thenReturn(MESSAGES);
+    Mockito.when(planMailResponseHelper.getWordCount(Mockito.any())).thenReturn(TEST_WORD_COUNT);
+    PlanMailResponse actual = getServletResponse();
+    PlanMailResponse expected = new PlanMailResponse(500, 50, 10, new ArrayList<>());
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void splitPotentialEvent() throws Exception {
+    // Test case where there are unread emails that will take 10 minutes to read
+    // and an event that splits the time interval required in two parts
+    // We should get 2 proposed events of 10 minutes each.
+    Mockito.when(calendarClient.getCalendarList()).thenReturn(ONE_CALENDAR);
+    Mockito.when(calendarClient.getUpcomingEvents(PRIMARY, CURRENT_TIME, END_TIME))
+        .thenReturn(EVENT_THREE);
+    Mockito.when(calendarClient.getCurrentTime()).thenReturn(CURRENT_TIME);
+    Mockito.when(gmailClient.getUnreadEmailsFromNDays(MESSAGE_FORMAT, NUMBER_DAYS_UNREAD))
+        .thenReturn(MESSAGES);
+    Mockito.when(planMailResponseHelper.getWordCount(Mockito.any())).thenReturn(TEST_WORD_COUNT);
+    PlanMailResponse actual = getServletResponse();
+    List<Date> startHour =
+        Arrays.asList(
+            new Date(2020 - OFFSET_YEAR, 4, 19, 10, 0),
+            new Date(2020 - OFFSET_YEAR, 4, 19, 10, 30));
+    List<Date> endHour =
+        Arrays.asList(
+            new Date(2020 - OFFSET_YEAR, 4, 19, 10, 5),
+            new Date(2020 - OFFSET_YEAR, 4, 19, 10, 35));
+    Assert.assertEquals(500, actual.getWordCount());
```

Comment:
This assert seems redundant, because you are testing the entire content of the actual response later.